tools/src/main/scala/scala/scalanative/codegen/AbstractCodeGen.scala

```diff
@@ -840,9 +1029,36 @@ private[codegen] abstract class AbstractCodeGen(
   private[codegen] def genCall(
       genBind: () => Unit,
       call: Op.Call,
-      unwind: Next
-  )(implicit fresh: Fresh, sb: ShowBuilder): Unit = {
+      unwind: Next,
+      pos: Option[Position],
+      scope: Option[Incr[LLVMDebugInformation.Scope]]
+  )(implicit
+      fresh: Fresh,
+      sb: ShowBuilder,
+      dwf: DebugInformationSection.Builder
+  ): Unit = {
     import sb._
+
+    val dwarfTok = scope
+      .map(subprogram =>
+        dwf.scopeLocation(pos.getOrElse(Position.NoPosition), subprogram)
+      )
+      .map(_.tok)
+
+    def addDebug() =
+      if (config.debugMetadata)
+        dwarfTok.foreach { tok =>
+          /** There are situations where the position is empty, for example in
+           *  situations where a null check is generated (and the function call
+           *  is throwNullPointer)
+           *
+           *  in this case we can only use NoPosition
+           */
```

Comment:
These functions are generated in the Lower, and we probably should be able to mitigate some of these problems in the future. Of course, we still can also get NoPosition from the compiler plugin, but that should be rare.